pytket/qir/conversion/api.py

```diff
@@ -46,6 +52,8 @@ def pytket_to_qir(
     wfh: Optional[wasm.WasmFileHandler] = None,
     int_type: int = 64,
     cut_pytket_register: bool = False,
+    compile_circuit: bool = False,
+    check_input: bool = True,
```

Comment:
I don't see the motivation for these arguments. The compliation pass is specific to H-series devices whereas this should eventually be a general QIR converter. What's wrong with the user compiling the circuit in whatever way they want before converting to QIR. Also I think the checks in `check_input` should happen in all cases; why would you want to turn them off?

Reply:
`check_input` is taking quiet long, if you are sure that your cirucit is working you might want to skip it.

Reply:
We had users complain about missing functionality to directly compiler circuit to a gateset which can be converted. The default of `compile_circuit` is false, so this is only done on request. If you think we don't need this, I am happy to remove it again.

Reply:
> `check_input` is taking quiet long, if you are sure that your cirucit is working you might want to skip it.

If `check_input` is false but some of the checks would fail, what happens? Does the user just get a conversion error further down the line, or does conversion succeed but produce invalid QIR? If the former, I'm fine with that; if the latter, I think that is an issue...

Reply:
> > `check_input` is taking quiet long, if you are sure that your cirucit is working you might want to skip it.
> 
> If `check_input` is false but some of the checks would fail, what happens? Does the user just get a conversion error further down the line, or does conversion succeed but produce invalid QIR? If the former, I'm fine with that; if the latter, I think that is an issue...

If the conversion works you will always get valid QIR. If the check would fail, it will fail later on in the conversion, it can sometimes happen that the error message is not as helpful as it would be when doing the check, but I think this is acceptable.

Reply:
> > We had users complain about missing functionality to directly compiler circuit to a gateset which can be converted. The default of `compile_circuit` is false, so this is only done on request. If you think we don't need this, I am happy to remove it again.
> 
> I would rather keep compilation separate, as it's a cleaner division of functions. Also having this option puts a greater testing burden on pytket-qir which is complex enough as it is.

Okay, I will remove this option.

Reply:
> > > `check_input` is taking quiet long, if you are sure that your cirucit is working you might want to skip it.
> > 
> > 
> > If `check_input` is false but some of the checks would fail, what happens? Does the user just get a conversion error further down the line, or does conversion succeed but produce invalid QIR? If the former, I'm fine with that; if the latter, I think that is an issue...
> 
> If the conversion works you will always get valid QIR. If the check would fail, it will fail later on in the conversion, it can sometimes happen that the error message is not as helpful as it would be when doing the check, but I think this is acceptable.

In that case, I'd recommend removing the option and relying on downstream errors.

Reply:
The compile option is now removed, see @cqc-melf
[remove compiler option from api](https://github.com/CQCL/pytket-qir/pull/147/commits/d83c45f7d794586d99beb86556802e1d9d30558c)
[d83c45f](https://github.com/CQCL/pytket-qir/pull/147/commits/d83c45f7d794586d99beb86556802e1d9d30558c)